### ISSUE TITLE
[Kura 2.1.0 QA] eth0 netmask configuration

### ISSUE THREAD

**Issue by markoer** [eclipse/kura#877]
2 similar cases are covered:

1. 16-bit netmask

Disable eth0 interface in Kura web UI.
Apply the configuration. Interface is not actually disabled (#876).
Now configure it as:
  * Enabled for WAN
  * Manually
  * add IP Address and Gateway IP
  * add netmask 255.255.0.0

Apply configuration.
As a result the actual applied netmask will be 255.255.255.0 and Kura's config will show 255.255.0.0.

2. 32-bit netmask

Disable and later configure eth0 as before.
This time set netmask to 255.255.255.255.
Apply configuration.
As a result the actual netmask will be 255.255.255.0 and Kura's config will show an empty field.

Note:
Netmask is properly applied only after IP Address is also updated. In case of 255.255.255.255 is is still not shown in web UI.

Kura 2.1.0 build 12, Raspberry Pi 3, fully updated `Linux raspberrypi 4.1.18-v7+ #846 SMP Thu Feb 25 14:22:53 GMT 2016 armv7l `

**Comment by cdealti** [comment]
1) It may take some time to apply the changes. Can you wait longer before you check the netmask with ``ip``

2) This is a corner case. I don't know of any such usage with Ethernet.

**Comment by markoer** [comment]
1. I find that `ifconfig` returns a bit nicer output. Anyway, the changes are normally updated rather fast - as soon as the UI closes the modal waiting indicator. And it seems also a second netmask update actually updates it. So - netmask is updated in case IP changes or netmask is updated the second time. If IP remains the same as before, netmask is not properly set the first time.

2. That's what test(er)s are for :)

**Comment by pierantoniomerlino** [comment]
1. 16-bit netmask

As reported in #867, the `dhcpcd` service is normally running on the RPi3. So, the netmask is updated by the dhcp server even if the user configured a static IP from Kura webUI. Removing (`apt-get purge dhcpcd5`) and killing the service fixes the issue.

2. 32-bit netmask

I agree that this is a corner case, however if the user sets a 255.255.255.255 netmask, the interface is accordingly configured and `ifconfig` reports the correct configuration. The problem is that the webUI doesn't show the netmask because this code on the TabTcpIpUi.java file:

```
        if (config != null && config.getSubnetMask() != null && config.getSubnetMask().equals("255.255.255.255")) {
            config.setSubnetMask("");
        }

```

**Comment by cdealti** [comment]
@markoer can we close this? As for #867 we will document to disable ``dhcpcd``.

About the /32 netmask not shown in the web UI I would create a separate issue with label ``kura-web``.

**Comment by markoer** [comment]
I retested the 16-bit scenario now that dhcpcd5 package is gone and it works as expected.
